ironfish/src/evm/evm.ts

```diff
@@ -39,4 +42,34 @@ export class IronfishEvm {
   async runTx(opts: RunTxOpts): Promise<RunTxResult> {
     return this.vm.runTx(opts)
   }
+
+  async verifyTx(opts: RunTxOpts): Promise<EvmResult> {
+    // TODO(jwp) add db transaction and roll back
+    const result = await this.runTx(opts)
+
+    // TODO(jwp) from custom opcodes populate shields and unshields
+    return {
+      result,
+      shields: [],
+      unshields: [],
+    }
+  }
+}
+
+type EvmShield = {
```

Comment:
Should we call it `mint` on the EVM side? I wonder if that will confuse people writing the contract. Maybe not, I just think shield/unshield is more intuitive